tot/plot_utils.py

```diff
@@ -0,0 +1,374 @@
+        xlabel : str
+            Label name on X-axis
+        ylabel : str
+            Label name on Y-axis
+        highlight_forecast : Union[int, None]
+            i-th step ahead forecast to highlight.
+        line_per_origin : bool
+            Print a line per forecast of one per forecast age
+        figsize : tuple [int, int]
+            Width, height in inches.
+
+    Returns
+    -------
+        Plotly figure
+    """
+    if resampler_active:
+        register_plotly_resampler(mode="auto")
+    else:
+        unregister_plotly_resampler()
+    cross_marker_color = "blue"
+    cross_symbol = "x"
+
+    fcst = fcst.fillna(value=np.nan)
+
+    ds = fcst["ds"].dt.to_pydatetime()
+    colname = "yhat"
+    step = 1
+    # all yhat column names, including quantiles
+    yhat_col_names = [col_name for col_name in fcst.columns if f"{colname}" in col_name]
+    # without quants
+    yhat_col_names_no_qts = [
+        col_name for col_name in yhat_col_names if f"{colname}" in col_name and "%" not in col_name
+    ]
+    data = []
+
+    if highlight_forecast is None:
+        for i, yhat_col_name in enumerate(yhat_col_names_no_qts):
+            data.append(
+                go.Scatter(
+                    name=yhat_col_name,
+                    x=ds,
+                    y=fcst[f"{colname}{i + 1}"],
+                    mode="lines",
+                    line=dict(color=f"rgba(45, 146, 255, {0.2 + 2.0 / (i + 2.5)})", width=line_width),
+                    fill="none",
+                )
+            )
+    if len(quantiles) > 1:
+        for i in range(1, len(quantiles)):
+            # skip fill="tonexty" for the first quantile
+            if i == 1:
+                data.append(
+                    go.Scatter(
+                        name=f"{colname}{highlight_forecast if highlight_forecast else step} {round(quantiles[i] * 100, 1)}%",
+                        x=ds,
+                        y=fcst[
+                            f"{colname}{highlight_forecast if highlight_forecast else step} {round(quantiles[i] * 100, 1)}%"
+                        ],
+                        mode="lines",
+                        line=dict(color="rgba(45, 146, 255, 0.2)", width=1),
+                        fillcolor="rgba(45, 146, 255, 0.2)",
+                    )
+                )
+            else:
+                data.append(
+                    go.Scatter(
+                        name=f"{colname}{highlight_forecast if highlight_forecast else step} {round(quantiles[i] * 100, 1)}%",
+                        x=ds,
+                        y=fcst[
+                            f"{colname}{highlight_forecast if highlight_forecast else step} {round(quantiles[i] * 100, 1)}%"
+                        ],
+                        mode="lines",
+                        line=dict(color="rgba(45, 146, 255, 0.2)", width=1),
+                        fill="tonexty",
+                        fillcolor="rgba(45, 146, 255, 0.2)",
```

Comment:
That is a lot of duplicate code to only change the `fill` (unless I missed something). Might be better to use the same code and only change the fill parameter.